chromadb/utils/data_loaders.py

```diff
@@ -22,3 +23,123 @@ def _load_image(self, uri: Optional[URI]) -> Optional[Image]:
     def __call__(self, uris: Sequence[Optional[URI]]) -> List[Optional[Image]]:
         with ThreadPoolExecutor(max_workers=self._max_workers) as executor:
             return list(executor.map(self._load_image, uris))
+
+# --------------------------- A lightweight util ---------------------------
+
+# TODO: Placing this here, since data loaders are a common use case for this function.
+#   But it should probably be moved to a more general-purpose module.
+#   See: https://github.com/chroma-core/chroma/issues/1606
+from functools import partial
+from concurrent.futures import ThreadPoolExecutor
+
+def vectorize(func, iterable=None, *, max_workers : int = 1):
+    """Like builtin map, but returns a list, 
+    and if iterable is None, returns a partial function that can directly be applied 
+    to an iterable.
+    This is useful, for instance, for making a data loader from any single-uri loader.
+    
+    Example:
+    >>> vectorize(lambda x: x**2, [1,2,3])
+    [1, 4, 9]
+    >>> vectorized_square = vectorize(lambda x: x**2)
+    >>> vectorized_square([1,2,3])
+    [1, 4, 9]
+    """
+    if iterable is None:
+        return partial(vectorize, func, max_workers=max_workers)
+    if max_workers == 1:
+        return list(map(func, iterable))
+    else:
+        with ThreadPoolExecutor(max_workers=max_workers) as executor:
+            return list(executor.map(func, iterable))
+        
+# --------------------------- Examples of loaders ---------------------------
+
+FileContents = Union[str, bytes]  # a type for the contents of a file
+
+def load_text(filepath: str) -> str:
+    with open(filepath, "r") as f:
+        return f.read()
+
+def load_bytes(filepath: str) -> bytes:
+    with open(filepath, "rb") as f:
+        return f.read()
+    
+def url_to_contents(
+        url: str, content_extractor=attrgetter('text'), *, params=None, **kwargs
+    ) -> FileContents:
+    import requests
+
+    response = requests.get(url, params=params, **kwargs)
+    response.raise_for_status()
+    return content_extractor(response)
+
+def pdf_file_text(
+        filepath: str, *, page_break_delim='---------------------------'
+    ) -> str:
+    from pypdf import PdfReader  # pip install pypdf
```

Comment:
Don't know what your policy is for local imports. The point being here that the function will work if, and only if, the package is installed, but won't oblige the whole project to need it. 

Perhaps, even better would be to import it via a context manager that will inform the user that they need to install pypdf to use this resource. The minimum effort way to do this is as I have done here: If the function call chokes, and the traceback is visible, the user will see the "pip install pypdf" comment and know what needs to be done. 

I really wanted to put this function here just to show a variety of loaders for `FileLoader`.

Reply:
Done.